hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -180,7 +180,11 @@ public boolean isBounded() {
         if (conf.getBoolean(FlinkOptions.READ_AS_STREAMING)) {
           StreamReadMonitoringFunction monitoringFunction = new StreamReadMonitoringFunction(
               conf, FilePathUtils.toFlinkPath(path), metaClient, maxCompactionMemoryInBytes);
-          OneInputStreamOperatorFactory<MergeOnReadInputSplit, RowData> factory = StreamReadOperator.factory((MergeOnReadInputFormat) getInputFormat(true));
+          InputFormat<RowData, ?> inputFormat = getInputFormat(true);
```

Comment:
would you please add an ut to verify the fix?

Reply:
ok , i will add some UT tomorrow

Reply:
You can add a IT in `HoodieDataSourceITCase`.